src/mcp_atlassian/confluence/pages.py

```diff
@@ -435,3 +437,38 @@ def delete_page(self, page_id: str) -> bool:
         except Exception as e:
             logger.error(f"Error deleting page {page_id}: {str(e)}")
             raise Exception(f"Failed to delete page {page_id}: {str(e)}") from e
+
+    def attach_content(
+        self, content: bytes, name: str, page_id: str
+    ) -> ConfluencePage | None:
+        """
+        Attach content to a Confluence page.
+
+        Args:
+            content: The content to attach (bytes)
+            name: The name of the attachment
+            page_id: The ID of the page to attach the content to
+
+        Returns:
+            ConfluencePage model containing the updated page's data or None if an error occurred
+        """
+        try:
+            logger.debug("Attaching content %s to page %s", name, page_id)
+            self.confluence.attach_content(content=content, name=name, page_id=page_id)
+        except ApiError as e:
+            logger.error(
+                "Confluence API Error when trying to attach content %s to page %s: %s",
+                name,
+                page_id,
+                str(e),
+            )
+            return None
```

Comment:
Returning `None` here means the `call_tool` function only gets a generic failure signal when this specific `ApiError` occurs.
Consider letting this `ApiError` propagate upwards by using `raise` instead of `return None`.